### ISSUE TITLE
[iOS] Bottom fixed container position shifts when presenting `modal` over `bottom tabs` (to be confirmed) 

### ISSUE THREAD

**Issue by guillaumehcht** [callstackincubator/react-native-bottom-tabs#224]
### Before submitting a new issue

- [X] I tested using the latest version of the library, as the bug might be already fixed.
- [X] I tested using a [supported version](https://github.com/reactwg/react-native-releases/blob/main/docs/support.md) of react native.
- [X] I checked for possible duplicate issues, with possible answers.

### Description
When presenting a modal screen on iOS **it seems that the Tab Bar is messing with the layout** in the modal **especially for things that are set to be absolutely positioned** at the bottom of the modal.

The bottom buttons container position keeps shifting between different Y coordinates (754px, 685px) causing visual glitches. This doesn't occur on Android.

> Is suspect it's coming from react-native-bottom-tabs because it is exactly 69px which usually corresponds to the height of the Tab Bar I believe.

I'm sorry if it's not coming from the package, I suspect it is but it might not be. I tried a lot of different solutions and I kept having this issue which was not there in previous packages I believe. (I'm gonna test to rollback to say 0.7.5 or 0.7.6 to check)

**Expected Behavior**
The button container should maintain a fixed position at the bottom of the screen without shifting.

**Actual Behavior**
The button container's Y position oscillates between 685px and 754px causing visual glitches.

**Additional Context**
- This only occurs on iOS
- The issue seems related to the interaction between the modal presentation and the bottom tabs
- The position shifts happen both on modal presentation and during scroll events

**The LOGS I'm getting:**
(NOBRIDGE) LOG  Button position changed: {"newY": 754, "previousY": 0, "timestamp": "2024-12-22T12:47:42.413Z"}
(NOBRIDGE) LOG  Button position changed: {"newY": 685, "previousY": 754, "timestamp": "2024-12-22T12:47:42.908Z"}
(NOBRIDGE) LOG  Button position changed: {"newY": 754, "previousY": 685, "timestamp": "2024-12-22T12:47:49.918Z"}

https://github.com/user-attachments/assets/a8f7f991-5dd0-4311-ba0b-3a2949422c78

### Library version

0.7.8

### Environment info

```shell
### Environment
- React Native: 0.76.5 (New Architecture)
- Expo SDK: ~52.0.20
- Platform: iOS
- Device: iPhone 14 Pro
- react-native-bottom-tabs: 0.7.8
- expo-router: ~4.0.14
```

### Steps to Reproduce
1. Create a modal screen with a ScrollView
2. Add a fixed position button container at the bottom
3. Open the modal from a screen with bottom tabs
4. Observe the button container position shifting after a couple of seconds on Modal mount

### Reproducible sample code

```js
### Code Example

// Bottom button container implementation
<View
  style={styles.buttonContainer}
  onLayout={(e) => {
    const { y } = e.nativeEvent.layout;
    if (y !== buttonLayout.y) {
      console.log("Button position changed:", {
        previousY: buttonLayout.y,
        newY: y,
        timestamp: new Date().toISOString(),
      });
      setButtonLayout({ y });
    }
  }}
>
  {/* Button content */}
</View>

// Styles
buttonContainer: {
  paddingBottom: 48,
  position: 'absolute',
  bottom: 0,
  left: 0,
  right: 0,
  justifyContent: 'center',
  alignItems: 'center',
}
```

**Comment by guillaumehcht** [comment]
Tested with 0.7.6 I still get the issue, maybe it's not the package. I'll continue trying to test other scenarios in the meantime.

**Comment by okwasniewski** [comment]
Hey @guillaumehcht, 

Sorry for the delay, I'm back to work after PTO 👍🏻 

Can you check if setting `contentInsetAdjustmentBehavior="never"` on the ScrollView inside of the modal fixes the issue? Probably it detects the bottom bar underneath the modal. Also are you using native modal?

**Comment by grabbou** [comment]
Thank you for opening this issue. 

Please provide reproducible example or code snippet demonstrating your use of the bottom sheet. There's a lot of different ways this can be done, want to make sure we can reproduce this.

Thank you,
Mike

**Comment by guillaumehcht** [comment]
Hey guys sorry I didn't saw the comments! I'll try to give you a repro soon! :)

(I did differently in the meantime as I couldn't get it to work so I forgot about this sorry)

**Comment by ChitrakshTarun** [comment]
Adding to this, there's an issue I noticed with the Keyboard opening over a FlatList on iOS, where a black box occupies the keyboard area while the keyboard scrolls from the bottom to top.

This issue does not occur when JS Tabs are used instead, or on Android with either setup.

Video reproduction 
- `0:00` to `0:08` is with `react-native-bottom-tabs`, where the black box appears when opening the keyboard
- `0:08` to `0:15` is with JS Tabs, where this issue does not occur.

Noticed a similar issue on this [tweet](https://x.com/iamarunabh/status/1883262699376877608) when you pause around `0:09` in the video, however slightly harder to pick in that video since the view background is black as well. 

The keyboard bug felt related to this issue so commenting it here. In case appropriate, please let me know and I will create a separate issue for the keyboard bug as well. Thank you!

https://github.com/user-attachments/assets/1b2d2220-52fe-456e-a114-544f0d9dfd09

**Comment by moritzlang** [comment]
I encountered the same problem a few days ago and created an [issue](https://github.com/expo/expo/issues/34352) in the expo repository (since I was not sure what was causing this bug).
Initially I thought that `<SafeAreaView />` does not work correctly, but after reading through this issue here it seems bottom tabs is indeed causing the position shifts inside the modal.

Detailed Issue in expo repository: https://github.com/expo/expo/issues/34352
Minimal reproducible example: https://github.com/moritzlang/expo-router-inconsistent-safe-area

As shown in the attached video, if you open the "modal" screen, the red floating button shifts its position between different Y coordinates.

https://github.com/user-attachments/assets/fb982a0e-01d0-4ec6-815c-63f82b66770f

**Comment by guillaumehcht** [comment]
Thanks @moritzlang for the repro!

Sorry didn't took the time to do it, was a bit under the water the last few weeks.

**Comment by okwasniewski** [comment]
Thanks for providing the repro @moritzlang, what's interesting is that you use JS Tabs from `expo-router` and you still get the issue which may seem like its not related to this library(!). Can you confirm that on the video you are using JS Tabs?

**Comment by moritzlang** [comment]
> Thanks for providing the repro [@moritzlang](https://github.com/moritzlang), what's interesting is that you use JS Tabs from `expo-router` and you still get the issue which may seem like its not related to this library(!). Can you confirm that on the video you are using JS Tabs?

@okwasniewski In the video [expo tabs](https://docs.expo.dev/router/advanced/tabs/) is used, which uses [Bottom Tabs Navigator](https://reactnavigation.org/docs/bottom-tab-navigator) from React Navigation under the hood.

**Comment by okwasniewski** [comment]
Thanks, so it looks like it's not related to this library if it happens within Expo Tabs / JS Tabs too. Does it happen without tabs at the bottom? (If you present the modal over a Stack navigator)

**Comment by moritzlang** [comment]
Yes, you are right. It also happens without the tabs. Thanks for your help!

https://github.com/user-attachments/assets/431ba8e7-33e7-48db-9ec8-13af68cc579e

**Comment by okwasniewski** [comment]
@guillaumehcht Would be awesome if you could confirm if the issue happens for you without tabs too. Then we can close the issue.

**Comment by guillaumehcht** [comment]
@okwasniewski just tested both scenario with JS Tabs and Native Tabs, the same flickering is happening. **So you were right, it's not happening in this library but most likely in Expo one!**

Sorry for opening that issue and testing it so late. 

**Thank you to all for having a look at that issue and reproducing it!** 

I'll close this issue now.
